### DISCUSSION TITLE
2024 Cohort 2 - Group 17 Discussion

### DISCUSSION THREAD

**Discussion by sharon-digipres** [Digital-Preservation-Coalition/PythonStudyGroups#43] [General]
This is the discussion thread for those in **Group 17** of the July-December 2024 Cohort

**Comment by emacglone** [comment]
Resources shared in our first session:

- https://github.com/bellingcat/open-source-research-notebooks
- https://glam-workbench.net/
- https://github.com/Lully
- https://github.com/Lully/bnf-sru
- https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/wiki/Additional-Resources
- https://www.py4e.com/
- https://programminghistorian.org/en/lessons/?topic=python
- https://programminghistorian.org/fr/lecons/?topic=python
- https://codeium.com/
- https://data.nls.uk/

**Comment by emacglone** [comment]
Hi all, a post to briefly record my learning over the last month. I have reviewed some of my work from [Python for Librarians](https://libraryjuiceacademy.com/shop/course/270-python-for-librarians/) (from a couple of years ago). I have reviewed some of my old notebooks and have decided that I want to create a notebook that takes columnar data and creates a node and edge tree diagram output.

**Reply by CRKussmann** [reply]
Thanks for your update.  Mine is similar.  I reviewed and followed most of the [Library Carpentry Python Intro for Libraries](https://librarycarpentry.org/lc-python-intro/) series.  When I got to the section about Using Pandas it helped me understand some code I had played with last session about comparing spreadsheet data.  I used the information in this section with some code checking/suggestions from ChatGTP to help me write a script in J.Notebook that takes a set of csv files (all with the same headers) and concatenates them all into one new csv.  Then I remove the rows in that new csv that do not start with a period (.) because I want to only view the file extensions (.pdf, .zip) rather than the rows that start with text like "Image formats" or "compressed formats".  This filtered view is then saved as another csv file which is mostly ready to be added to my Tableau datasource.  

I still want to use what is in that Panda's section to add some additional data to the spreadsheets prior to being concatenated.

**Comment by evacmueller** [comment]
Hi everyone. In the last month I played with some of the notebooks from [Bellingcat](https://github.com/bellingcat/open-source-research-notebooks). I didn't create my own but want to try for the next month with the help of the [Library Carpentry](https://librarycarpentry.org/lc-python-intro) workshop

**Comment by evacmueller** [comment]
Also while it's not specifically about Notebooks, I found two general Python/Programming Youtube channels that do really good explanations of various concepts:
[ArjanCodes](https://www.youtube.com/@ArjanCodes/videos)
[OOP](https://www.youtube.com/watch?v=ZDa-Z5JzLYM&list=PL-osiE80TeTsqhIuOqKhwlXsIBIdSeYtc&index=2)
